include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -794,6 +989,56 @@ class operational_domain_impl
         // if we made it here, the layout is operational
         return operational();
     }
+    /**
+     * This function checks if the given charge distribution surface (CDS) is physically valid for the parameter point
+     * represented by the step point `sp`.
+     *
+     * @param lyt CDS to check.
+     * @param sp Step point to be investigated.
+     * @return The operational status of the layout under the given simulation parameters.
+     */
+    operational_status is_step_point_suitable(Lyt& lyt, const step_point& sp) noexcept
+    {
+        // if the point has already been sampled, return the stored operational status
+        if (const auto op_value = has_already_been_sampled(sp); op_value.has_value())
+        {
+            return *op_value;
+        }
+
+        // fetch the x and y dimension values
+        const auto param_point = to_parameter_point(sp);
+
+        const auto operational = [this, &param_point]()
+        {
+            op_domain.operational_values[param_point] = operational_status::OPERATIONAL;
+
+            return operational_status::OPERATIONAL;
+        };
+
+        const auto non_operational = [this, &param_point]()
+        {
+            op_domain.operational_values[param_point] = operational_status::NON_OPERATIONAL;
+
+            return operational_status::NON_OPERATIONAL;
+        };
+
+        // increment the number of evaluated parameter combinations
+        ++num_evaluated_parameter_combinations;
+
+        sidb_simulation_parameters sim_params = params.simulation_parameters;
+        set_x_dimension_value(sim_params, param_point.x);
+        set_y_dimension_value(sim_params, param_point.y);
+
+        lyt.assign_physical_parameters(sim_params);
+
+        if (lyt.is_physically_valid())
+        {
+            return operational();
+        }
+
+        // if we made it here, the layout is operational
+        return non_operational();
```

Comment:
Sorry, I missed this in the previous review. Something seems not right here. Either the comment or the code is wrong.

Reply:
sorry, you are right! This is a typo.